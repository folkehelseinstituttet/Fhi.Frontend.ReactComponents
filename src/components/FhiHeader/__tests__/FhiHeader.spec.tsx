import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../FhiHeader';

describe('FhiHeader', () => {
  test('snapshot test', () => {
    const header = renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <Header
          className="test"
          projectName="Testproject"
          logo="../assets/logo-norsk-kun-forkortelse-svart-uten-luft-rundt.svg"
          logoLight="../assets/logo-norsk-kun-forkortelse-hvit-uten-luft-rundt.svg"
          menuItems={[
            {
              name: 'Dette er en tittel',
              routerLink: '/rl',
            },
          ]}
          faIcons={{
            menu: faBars,
            close: faTimes,
          }}
        />
      </MemoryRouter>,
    );
    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
