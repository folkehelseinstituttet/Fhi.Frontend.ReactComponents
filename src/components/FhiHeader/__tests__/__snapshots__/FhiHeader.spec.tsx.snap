// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FhiHeader snapshot test 1`] = `
<header
  className="FhiHeader__PaddingHeader-sc-zidcyl-0 dLwaaV fhi-app__header test"
>
  <div
    className="fhi-app__brand"
  >
    <div
      className="container fhi-app__brand-container"
    >
      <div
        className="fhi-app__brand-content"
      >
        <a
          className="fhi-app__logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="FHI-logo"
            src="../assets/logo-norsk-kun-forkortelse-svart-uten-luft-rundt.svg"
          />
        </a>
        <div
          className="fhi-app__project"
        >
          <span
            className="fhi-app__project-name"
          >
            Testproject
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="fhi-app__main-menu"
  >
    <div
      className="container fhi-app__main-menu-container"
    >
      <nav
        className="fhi-main-menu"
      >
        <div
          className="container fhi-main-menu__container"
        >
          <a
            className="fhi-main-menu__home-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="FHI-logo"
              className="fhi-main-menu__home-link-img"
              src="../assets/logo-norsk-kun-forkortelse-hvit-uten-luft-rundt.svg"
            />
          </a>
          <button
            aria-controls="fhi-main-menu"
            aria-expanded={false}
            className=" fhi-main-menu__toggler"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bars "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={{}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                fill="currentColor"
                style={{}}
              />
            </svg>
            <span
              className="fhi-main-menu__toggler-text"
            >
              Meny
            </span>
          </button>
          <div
            className="collapse fhi-main-menu__collapse"
          >
            <ul
              className="nav flex-column flex-xl-row fhi-main-menu__nav"
            >
              <li
                className="nav-item"
              >
                <a
                  className="nav-link fhi-main-menu__nav-link"
                  href="/rl"
                  onClick={[Function]}
                >
                  <span
                    className="fhi-main-menu__nav-link-text"
                  >
                    Dette er en tittel
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </div>
</header>
`;
